### PR TITLE
Internal: General tweaks for checklist module [ED-15738]

### PR DESCRIPTION
**Pull request by ronenboxer**

### CODE REVIEW

File: modules/checklist/assets/js/app/components/checklist-card-content.js [hunk 1]
```diff
@@ -1,6 +1,10 @@
 import { Button, Card, CardActions, CardContent, CardMedia, Link, Typography } from '@elementor/ui';
 import { __ } from '@wordpress/i18n';
 import PropTypes from 'prop-types';
+import { getAndUpdateStep } from '../../utils/functions';
+import { STEP } from '../../utils/consts';
+
+const { IS_MARKED_COMPLETED, IS_ABSOLUTE_COMPLETED, IS_IMMUTABLE_COMPLETED } = STEP;
```
Comment (davseve) [inline modules/checklist/assets/js/app/components/checklist-card-content.js:7]: Maybe destructure this in one line?
Not crucial.

File: modules/checklist/assets/js/app/components/header.js [hunk 2]
```diff
@@ -2,45 +2,35 @@ import { Typography, CloseButton, AppBar, Divider, Toolbar } from '@elementor/ui
...
-		if ( closedForFirstTime !== true ) {
-			await fetch( `${ elementorCommon.config.urls.rest }elementor/v1/checklist/user-progress`, {
-				method: 'POST',
-				headers: {
-					'Content-Type': 'application/json',
-					'X-WP-Nonce': elementorWebCliConfig.nonce,
-				},
+		if ( ! closedForFirstTime ) {
+			await $e.data.update( 'checklist/user-progress', {
+				[ FIRST_CLOSED_CHECKLIST_IN_EDITOR ]: true,
```
Comment (davseve) [inline modules/checklist/assets/js/app/components/header.js:27]: Maybe?
```suggestion
				[ CHECKLIST_CLOSED_IN_THE_EDITOR_FOR_FIRST_TIME ]: true,
```
From this text we can understand what this const means.

File: modules/checklist/assets/js/topbar-icon.js [hunk 3]
```diff
@@ -1,32 +1,27 @@
...
+import { useQuery } from '@elementor/query';
 import { __privateListenTo as listenTo, commandEndEvent } from '@elementor/editor-v1-adapters';
 import RocketIcon from '@elementor/icons/RocketIcon';
 import { Infotip } from '@elementor/ui';
-import ReminderModal from './components/reminder-modal';
+import ReminderModal from './app/components/reminder-modal';
 import * as React from 'react';
+import { USER_PROGRESS } from './utils/consts';
+
+const { FIRST_CLOSED_CHECKLIST_IN_EDITOR } = USER_PROGRESS;
```
Comment (davseve) [inline modules/checklist/assets/js/topbar-icon.js:10]: Same as above

File: modules/checklist/assets/js/app/components/checklist-card-content.js (hunk 1 above)
Comment (ronenelementor) [reply modules/checklist/assets/js/app/components/checklist-card-content.js:7]: I tried, doesn't work inside an import AFAIK

File: modules/checklist/assets/js/app/components/header.js (hunk 2 above)
Comment (ronenelementor) [reply modules/checklist/assets/js/app/components/header.js:27]: Sure!

File: modules/checklist/assets/js/topbar-icon.js (hunk 3 above)
Comment (ronenelementor) [reply modules/checklist/assets/js/topbar-icon.js:10]: [Crying emoticon]

**Review by louiswol94 — APPROVED** [review]

**Pull request merged**